### PR TITLE
Add core files.

### PR DESCRIPTION
**Pull request by mattcollier**
This is a WIP.

@msporny I believe this is how we do consensus storage properly.  See TODO about extending storage classes.

Depends on 

1. https://github.com/digitalbazaar/bedrock-ledger-node/pull/37
2. https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/pull/126

### CODE REVIEW

**Comment by dlongley** [comment]
@mattcollier,

Why is this approach better than just doing the extension in the consensus plugin itself?

**Comment by mattcollier** [comment]
Are you asking, 'Why have a separate module?' or 'Why put custom indexing and interactions with the storage layer in a plugin?'

**Comment by dlongley** [comment]
@mattcollier,

Also, I realize we only have a mongodb storage plugin right now, but should these sorts of plugins also call out that they only work with a particular master storage plugin?

**Comment by dlongley** [comment]
@mattcollier,

> Are you asking, 'Why have a separate module?' or 'Why put custom indexing and interactions with the storage layer in a plugin?'

Both. :)

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+*.log
+*.py[co]
```
Comment (davidlehn) [inline .gitignore:2]: Remove ;-)

**Comment by mattcollier** [comment]
I think it makes sense overall.  But mainly, I would like to see a reduction in the size of the continuity code base.  I also would like to have *VERY* tight version controls on the indexes.  I don't want any index changes slipping in unnoticed (ever!).  And lastly, I agree with Manu's observation that Continuity is very tightly coupled with the mongo storage, this is a step towards using the plugin interface to address that.

File: .gitignore (hunk 1 above)
Comment (mattcollier) [reply .gitignore:2]: you never know man, you never know.

**Review by dlongley — APPROVED** [review]

**Comment by mattcollier** [comment]
@dlongley you make a good point about explicit associations between plugins and storage layer.  Create and issue: https://github.com/digitalbazaar/bedrock-ledger-node/issues/38

**Pull request merged**